arci/src/clients/lazy.rs

```diff
@@ -0,0 +1,158 @@
+use std::{
+    sync::Arc,
+    time::{Duration, SystemTime},
+};
+
+use async_trait::async_trait;
+use nalgebra::{Isometry2, Isometry3};
+use tracing::error;
+
+use crate::{
+    error::Error,
+    gamepad::GamepadEvent,
+    traits::{
+        BaseVelocity, Gamepad, JointTrajectoryClient, Localization, MoveBase, Navigation, Speaker,
+        TrajectoryPoint, TransformResolver,
+    },
+    waits::WaitFuture,
+};
+
+#[allow(clippy::type_complexity)]
+pub struct Lazy<'a, T>(
+    once_cell::sync::Lazy<
+        Result<T, Arc<Error>>,
+        Box<dyn FnOnce() -> Result<T, Arc<Error>> + Send + Sync + 'a>,
+    >,
+);
+
+impl<'a, T> Lazy<'a, T> {
+    pub fn new(f: impl FnOnce() -> Result<T, Error> + Send + Sync + 'a) -> Self {
+        Self(once_cell::sync::Lazy::new(Box::new(|| {
+            f().map_err(Arc::new)
+        })))
+    }
+
+    pub fn get_ref(&self) -> Result<&T, Error> {
+        self.0.as_ref().map_err(|e| Error::Arc(e.clone()))
+    }
+}
+
+impl<T> JointTrajectoryClient for Lazy<'_, T>
+where
+    T: JointTrajectoryClient,
+{
+    fn joint_names(&self) -> Vec<String> {
+        match self.get_ref() {
+            Ok(this) => this.joint_names(),
+            Err(e) => {
+                error!("{}", e);
+                // TODO
+                vec![]
+            }
+        }
+    }
```

Comment:
Perhaps we need to change `joint_names` to return `Result`?

Reply:
Are you mention about changing arci::JointTrajectoryClient::joint_names(), right?
I think basically, it never fails.

Reply:
Changed to return Result if Result can be returned, and to panic if not.